test.js

```diff
@@ -7,668 +7,465 @@ const Next = require('next')
-    method: 'GET'
-  }, (err, res) => {
-    t.error(err)
-    t.equal(res.statusCode, 200)
-    t.equal(res.headers['content-type'], 'text/html; charset=utf-8')
-  })
+  fastify.close()
 })
 
-test('should support different methods', t => {
-  t.plan(3)
+test('should support HEAD method', async t => {
+  const fastify = await Fastify().register(require('./index'))
+  fastify.next('/hello', { method: 'head' })
 
-  const fastify = Fastify()
-  t.teardown(() => fastify.close())
+  const origin = await fastify.listen({ port })
+  const { statusCode, headers } = await request({ path: '/hello', origin, method: 'HEAD' })
 
-  fastify
-    .register(require('./index'))
-    .after(() => {
-      fastify.next('/hello', { method: 'options' })
-    })
+  t.equal(statusCode, 200)
+  t.equal(headers['content-type'], 'text/html; charset=utf-8')
 
-  fastify.inject({
-    url: '/hello',
-    method: 'OPTIONS'
-  }, (err, res) => {
-    t.error(err)
-    t.equal(res.statusCode, 200)
-    t.equal(res.headers['content-type'], 'text/html; charset=utf-8')
-  })
+  fastify.close()
 })
 
-test('should support a custom handler', t => {
-  t.plan(3)
+test('should support a custom handler', async t => {
+  const fastify = await Fastify().register(require('./index'))
+  fastify.next('/hello', (app, req, reply) => {
+    app.render(req.raw, reply.raw, '/hello', req.query, {})
+  })
 
-  const fastify = Fastify()
-  t.teardown(() => fastify.close())
+  const origin = await fastify.listen({ port })
+  const { statusCode, headers } = await request({ path: '/hello', origin })
 
-  fastify
-    .register(require('./index'))
-    .after(() => {
-      fastify.next('/hello', (app, req, reply) => {
-        app.render(req.raw, reply.raw, '/hello', req.query, {})
-      })
-    })
+  t.equal(statusCode, 200)
+  t.equal(headers['content-type'], 'text/html; charset=utf-8')
 
-  fastify.inject({
-    url: '/hello',
-    method: 'GET'
-  }, (err, res) => {
-    t.error(err)
-    t.equal(res.statusCode, 200)
-    t.equal(res.headers['content-type'], 'text/html; charset=utf-8')
-  })
+  fastify.close()
 })
 
-test('should return 404 on undefined route', t => {
-  t.plan(2)
+test('should return 404 on undefined route', async t => {
+  const fastify = await Fastify().register(require('./index'))
+  fastify.next('/hello')
 
-  const fastify = Fastify()
-  t.teardown(() => fastify.close())
+  const origin = await fastify.listen({ port })
+  const { statusCode } = await request({ path: '/test', origin })
 
-  fastify
-    .register(require('./index'))
-    .after(() => {
-      fastify.next('/hello')
-    })
+  t.equal(statusCode, 404)
 
-  fastify.inject({
-    url: '/test',
-    method: 'GET'
-  }, (err, res) => {
-    t.error(err)
-    t.equal(res.statusCode, 404)
-  })
+  fastify.close()
 })
 
-test('should throw if path is not a string', t => {
-  t.plan(2)
+test('should throw if path is not a string', async t => {
+  const fastify = await Fastify().register(require('./index'))
```

Comment:
nitpick:
maybe would be nicer to require only once ./index as fastifyNext and only pass it every where instead of doing again and again require.

Reply:
Yes, I thought of that and even running `fastify.close()` after each test, but as a personal choice I like for every tests to be as encapsulated as possible for the sake of clearness and simplicity. I made an exception with the `testNextAsset` function basically because it was already there.

Reply:
thats not what I meant. I would have rejected if we would have used only one fastify instance. 
What i meant was to import at the head of the file

```
const fastifyNext = require('./index')
```

and then here you would call

```
  const fastify = await Fastify().register(fastifyNext)
```

Reply:
Oh, I see, sorry. Yes, that makes perfect sense and I like it, will push a change with that, thanks!

Reply:
Done. Please check again and resolve if you're ok with the changes.